cmd/minishift/cmd/root.go

```diff
@@ -213,11 +222,34 @@ func initializeProfile() string {
 		activeProfile string
 	)
 
+	// If *minishift profile set* or *minishift start --profile* command is used.
 	for i, arg := range os.Args {
-		if arg == "--"+profileFlag {
-			profileName = os.Args[i+1]
-			if !cmdUtil.IsValidProfileName(profileName) {
-				atexit.ExitWithMessage(1, "Profile names must consist of alphanumeric characters only.")
+		// This will match if we used *profile* flag
+		if arg == profileFlag {
+			// This make sure if user specify *profile* flag then
+			// it should not panic with out of index error
+			if len(os.Args) <= i+2 {
+				break
+			}
+			// This will match if after *profile* we used *set* command
+			// Then it will have profileName as whatever we provided as set command parameter.
+			if os.Args[i+1] == profileSetFlag {
+				profileName = os.Args[i+2]
```

Comment:
We have the same code in initializeProfile() i.e. 
```
if os.Args[i+1] == profileSetFlag {
				profileName = os.Args[i+2]
```
We should create a function for this and use it both the places.

Reply:
@LalatenduMohanty This is still `initializeProfile()` function code only where you think this logic is used again?

Reply:
@praveenkumar nvm.